<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>“Moving Pictures” tutorial &#8212; QIIME 2 2.0.5 documentation</title>
    
    <link rel="stylesheet" href="../_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '2.0.5',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="top" title="QIIME 2 2.0.5 documentation" href="../index.html" />
    <link rel="up" title="Tutorials" href="index.html" />
    <link rel="next" title="Fecal microbiota transplant (FMT) study: an exercise" href="fmt.html" />
    <link rel="prev" title="Tutorials" href="index.html" />
   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head>
  <body role="document">
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="moving-pictures-tutorial">
<h1>&#8220;Moving Pictures&#8221; tutorial<a class="headerlink" href="#moving-pictures-tutorial" title="Permalink to this headline">¶</a></h1>
<p>In this tutorial you&#8217;ll use QIIME 2 to perform an analysis of human microbiome samples from two individuals at four body sites at five timepoints, the first of which immediately followed antibiotic usage. A study based on these samples was originally published in <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/21624126">Caporaso et al. (2011)</a>. The data used in this tutorial were sequenced on an Illumina HiSeq using the <a class="reference external" href="http://earthmicrobiome.org">Earth Microbiome Project</a> hypervariable region 4 (V4) 16S rRNA sequencing protocol.</p>
<p>You should first look through the sample metadata to familiarize yourself with the samples used in this study. The <a class="reference external" href="https://docs.google.com/spreadsheets/d/1FXHtTmvw1gM4oUMbRdwQIEOZJlhFGeMNUvZmuEFqpps/edit?usp=sharing">sample metadata</a> is available as a Google Spreadsheet. You should download this file as tab-separated text by selecting <code class="docutils literal"><span class="pre">File</span></code> &gt; <code class="docutils literal"><span class="pre">Download</span> <span class="pre">as</span></code> &gt; <code class="docutils literal"><span class="pre">Tab-separated</span> <span class="pre">values</span></code>. Save the file as <code class="docutils literal"><span class="pre">sample-metadata.tsv</span></code>.</p>
<div class="qiime1 admonition" id="qiime1users-0">
<p class="first admonition-title">QIIME 1 Users</p>
<p class="last">These are the same data that are used in the QIIME 1 <a class="reference external" href="http://nbviewer.jupyter.org/github/biocore/qiime/blob/1.9.1/examples/ipynb/illumina_overview_tutorial.ipynb">Illumina Overview Tutorial</a>.</p>
</div>
<div class="section" id="prepare-for-the-analysis">
<h2>Prepare for the analysis<a class="headerlink" href="#prepare-for-the-analysis" title="Permalink to this headline">¶</a></h2>
<p>First create a new directory and change to that directory.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>mkdir qiime2-moving-pictures-tutorial
<span class="nb">cd</span> qiime2-moving-pictures-tutorial
</pre></div>
</div>
<p>Then, download the raw sequences that we&#8217;ll use in this analysis. In this tutorial we&#8217;ll work with a small subset of the complete sequence data so that the commands will run quickly.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>mkdir raw-sequences
curl -sL http://data.qiime2.org/tutorials/moving-pictures/raw-sequences/barcodes.fastq.gz &gt; raw-sequences/barcodes.fastq.gz
curl -sL http://data.qiime2.org/tutorials/moving-pictures/raw-sequences/sequences.fastq.gz &gt; raw-sequences/sequences.fastq.gz
</pre></div>
</div>
<p>All data that is used as input to QIIME 2 is in form of QIIME 2 artifacts, which contain information about the type of data and the source of the data. So, the first thing we need to do is import these raw data files into a QIIME 2 artifact. The semantic type of this artifact is <code class="docutils literal"><span class="pre">RawSequences</span></code>.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime tools import --type RawSequences --input-path raw-sequences/ --output-path raw-sequences.qza
</pre></div>
</div>
<p><code class="docutils literal"><span class="pre">RawSequences</span></code> artifacts contain sequences that are multiplexed, meaning that the sequences have not yet been assigned to samples (hence the inclusion of both <code class="docutils literal"><span class="pre">sequences.fastq.gz</span></code> and <code class="docutils literal"><span class="pre">barcodes.fastq.gz</span></code> files, where the <code class="docutils literal"><span class="pre">barcodes.fastq.gz</span></code> contains the barcode (i.e., index) read associated with each sequence in <code class="docutils literal"><span class="pre">sequences.fastq.gz</span></code>.)</p>
<div class="qiime1 admonition" id="qiime1users-1">
<p class="first admonition-title">QIIME 1 Users</p>
<p class="last">In QIIME 1, we generally suggested performing demultiplexing through QIIME (e.g., with <code class="docutils literal"><span class="pre">split_libraries.py</span></code> or <code class="docutils literal"><span class="pre">split_libraries_fastq.py</span></code>) as this step also performed quality control of sequences. We now separate the demultiplexing and quality control steps, so you can begin QIIME 2 with either multiplexed sequences (as we&#8217;re doing here) or demultiplexed sequences.</p>
</div>
</div>
<div class="section" id="demultiplexing-sequences">
<h2>Demultiplexing sequences<a class="headerlink" href="#demultiplexing-sequences" title="Permalink to this headline">¶</a></h2>
<p>To demultiplex sequences we need to know which barcode sequence is associated with each sample. This information is contained in the <a class="reference external" href="https://docs.google.com/spreadsheets/d/1FXHtTmvw1gM4oUMbRdwQIEOZJlhFGeMNUvZmuEFqpps/edit?usp=sharing">sample metadata</a> file, so you should download that file if you haven&#8217;t already. You can then run the following command to demultiplex the sequences (the <code class="docutils literal"><span class="pre">demux</span> <span class="pre">emp</span></code> command refers to the fact that these sequences are barcoded according to the <a class="reference external" href="http://earthmicrobiome.org">Earth Microbiome Project</a> protocol). This will result in a new <code class="docutils literal"><span class="pre">SampleData[SequencesWithQuality]</span></code> artifact, where each sequence read is associated with its source sample.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime demux emp --i-seqs raw-sequences.qza --m-barcodes-file sample-metadata.tsv --m-barcodes-category BarcodeSequence --o-per-sample-sequences demux
</pre></div>
</div>
</div>
<div class="section" id="sequence-quality-control">
<h2>Sequence quality control<a class="headerlink" href="#sequence-quality-control" title="Permalink to this headline">¶</a></h2>
<p>We&#8217;ll next perform quality control on the demultiplexed sequences using <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/27214047">DADA2</a>. DADA2 is a pipeline for detecting and correcting (where possible) Illumina amplicon sequence data. As implemented in the <code class="docutils literal"><span class="pre">q2-dada2</span></code> plugin, this quality control process will additionally filter any phiX reads (a common experimental artifact) that are identified in the sequencing data, and will filter chimeric sequences. The result of this step will be a <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> artifact, which contains counts (frequencies) of each unique sequence in each sample in the dataset, and a <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> artifact, which maps feature identifiers in the <code class="docutils literal"><span class="pre">FeatureTable</span></code> to the sequences they represent.</p>
<div class="qiime1 admonition" id="qiime1users-2">
<p class="first admonition-title">QIIME 1 Users</p>
<p class="last">The <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> artifact is the equivalent of the QIIME 1 OTU or BIOM table, and the <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> artifact is the equivalent of the QIIME 1 <em>representative sequences</em> file. Because the &#8220;OTUs&#8221; resulting from DADA2 are creating by grouping unique sequences, these are the equivalent of 100% OTUs from QIIME 1. In DADA2, these 100% OTUs are referred to as <em>denoised sequence variants</em>. In QIIME 2, these OTUs are higher resolution than the QIIME 1 default of 97% OTUs, and they&#8217;re higher quality due to the DADA2 denoising process. This should therefore result in more accurate estimates of diversity and taxonomic composition of samples than was achieved with QIIME 1.</p>
</div>
<p>The <code class="docutils literal"><span class="pre">dada2</span> <span class="pre">denoise</span></code> method requires two parameters that are used in quality filtering: <code class="docutils literal"><span class="pre">--p-trim-left</span> <span class="pre">m</span></code>, which trims off the first <code class="docutils literal"><span class="pre">m</span></code> bases of each sequence, and <code class="docutils literal"><span class="pre">--p-trunc-len</span> <span class="pre">n</span></code> which truncates each sequence at position <code class="docutils literal"><span class="pre">n</span></code>. This allows the user to remove low quality regions of the sequences. To determine what values to pass for these two parameters, you should first run the <code class="docutils literal"><span class="pre">dada2</span> <span class="pre">plot-qualities</span></code> visualizer, which will generate plots of the quality scores by position for a randomly selected set of samples. In the following command, we&#8217;ll generate a quality plot using 10 randomly selected samples (specified by passing <code class="docutils literal"><span class="pre">--p-n</span> <span class="pre">10</span></code>). Run the following command and view the resulting visualization.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">All QIIME 2 visualizers (i.e., commands that take a <code class="docutils literal"><span class="pre">--o-visualization</span></code> parameter) will generate a <code class="docutils literal"><span class="pre">.qzv</span></code> file. You can view these files with <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">tools</span> <span class="pre">view</span></code>. We provide the command to view this first visualization, but for the remainder of this tutorial we&#8217;ll tell you to <em>view the resulting visualization</em> after running a visualizer, which means that you should run <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">tools</span> <span class="pre">view</span></code> on the .qzv file that was generated.</p>
</div>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime dada2 plot-qualities --i-demultiplexed-seqs demux.qza --o-visualization demux-qual-plots --p-n 10

qiime tools view demux-qual-plots.qzv
</pre></div>
</div>
<p><strong>demux-qual-plots</strong>: <a class="reference external" href="../../_viz/demux-qual-plots/df384304-e414-4669-93a4-5ee997723b21/data/index.html">visualization</a>, <a class="reference external" href="../../_viz/demux-qual-plots.qzv">qzv</a></p>
<div class="question admonition" id="question-0">
<p class="first admonition-title">Question</p>
<p class="last">Based on the plots you see in <code class="docutils literal"><span class="pre">demux-qual-plots.qzv</span></code>, what values would you choose for <code class="docutils literal"><span class="pre">--p-trunc-len</span></code> and <code class="docutils literal"><span class="pre">--p-trim-left</span></code> in this case?</p>
</div>
<p>In these plots, the quality of the initial bases seems to be high, so we won&#8217;t trim any bases from the beginning of the sequences. The quality seems to drop off around position 100, so we&#8217;ll truncate our sequences at 100 bases. This next command may take up to 10 minutes to run, and is the slowest step in this tutorial.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime dada2 denoise --i-demultiplexed-seqs demux.qza --p-trim-left <span class="m">0</span> --p-trunc-len <span class="m">100</span> --o-representative-sequences rep-seqs --o-table table
</pre></div>
</div>
<p>After the <code class="docutils literal"><span class="pre">dada2</span> <span class="pre">denoise</span></code> step completes, you&#8217;ll want to explore the resulting data. You can do this using the following two commands, which will create visual summaries of the data. The <code class="docutils literal"><span class="pre">feature-table</span> <span class="pre">summarize</span></code> command will give you information on how many sequences are associated with each sample and with each feature, histograms of those distributions, and some related summary statistics. The <code class="docutils literal"><span class="pre">feature-table</span> <span class="pre">view-seq-data</span></code> will provide a mapping of feature IDs to sequences, and provide links to easily BLAST each sequence against the NCBI nt database. The latter visualization will be very useful later in the tutorial, when you want to learn more about specific features that are important in the data set.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime feature-table summarize --i-table table.qza --o-visualization table
qiime feature-table view-seq-data --i-data rep-seqs.qza --o-visualization rep-seqs
</pre></div>
</div>
<p><strong>table</strong>: <a class="reference external" href="../../_viz/table/19d6165e-e76a-4a87-85c5-b270494f9357/data/index.html">visualization</a>, <a class="reference external" href="../../_viz/table.qzv">qzv</a></p>
<p><strong>rep-seqs</strong>: <a class="reference external" href="../../_viz/rep-seqs/f6946a20-e6ba-46c2-8f54-c8333a4c05b4/data/index.html">visualization</a>, <a class="reference external" href="../../_viz/rep-seqs.qzv">qzv</a></p>
</div>
<div class="section" id="generate-a-tree-for-phylogenetic-diversity-analyses">
<h2>Generate a tree for phylogenetic diversity analyses<a class="headerlink" href="#generate-a-tree-for-phylogenetic-diversity-analyses" title="Permalink to this headline">¶</a></h2>
<p>QIIME supports several phylogenetic diversity metrics, including Faith&#8217;s Phylogenetic Diversity and weighted and unweighted UniFrac. In addition to counts of features per sample (i.e., the data in the <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> artifact), these metrics require a rooted phylogenetic tree relating the features to one another. This information will be stored in a <code class="docutils literal"><span class="pre">Phylogeny[Rooted]</span></code> artifact. The following steps will generate this artifact.</p>
<p>First, we perform a multiple sequence alignment of the sequences in our <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> to create a <code class="docutils literal"><span class="pre">FeatureData[AlignedSequence]</span></code> artifact. Here we do this with the <cite>mafft</cite> program.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime alignment mafft --i-sequences rep-seqs.qza --o-alignment aligned-rep-seqs
</pre></div>
</div>
<p>Next, we mask (or filter) the alignment to remove positions that are highly variable. These positions are generally considered to add noise to a resulting phylogenetic tree.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime alignment mask --i-alignment aligned-rep-seqs.qza --o-masked-alignment masked-aligned-rep-seqs
</pre></div>
</div>
<p>Next, we&#8217;ll apply FastTree to generate a phylogenetic tree from the masked alignment.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime phylogeny fasttree --i-alignment masked-aligned-rep-seqs.qza --o-tree unrooted-tree
</pre></div>
</div>
<p>The FastTree program creates an unrooted tree, so in the final step in this section we apply midpoint rooting to place the root of the tree at the midpoint of the longest tip-to-tip distance in the unrooted tree.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime phylogeny midpoint-root --i-tree unrooted-tree.qza --o-rooted-tree rooted-tree
</pre></div>
</div>
</div>
<div class="section" id="alpha-and-beta-diversity-analysis">
<h2>Alpha and beta diversity analysis<a class="headerlink" href="#alpha-and-beta-diversity-analysis" title="Permalink to this headline">¶</a></h2>
<p>QIIME 2&#8217;s diversity analyses are available through the <code class="docutils literal"><span class="pre">q2-diversity</span></code> plugin, which supports computing alpha and beta diversity metrics, applying related statistical tests, and generating interactive visualizations. We&#8217;ll first apply the <code class="docutils literal"><span class="pre">core-metrics</span></code> method, which rarefies a <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> to a user-specified depth, and then computes a series of alpha and beta diversity metrics. The metrics computed by default are:</p>
<ul class="simple">
<li>Alpha diversity</li>
</ul>
<blockquote>
<div><ul class="simple">
<li>Shannon&#8217;s diversity index (a quantitative measure of community richness)</li>
<li>Observed OTUs (a qualitative measure of community richness)</li>
<li>Faith&#8217;s Phylogenetic Diversity (a qualitiative measure of community richness that incorporates phylogenetic relationships between the features)</li>
<li>Evenness (or Pielou&#8217;s Evenness; a measure of community evenness)</li>
</ul>
</div></blockquote>
<ul class="simple">
<li>Beta diversity</li>
</ul>
<blockquote>
<div><ul class="simple">
<li>Jaccard distance (a qualitative measure of community dissimilarity)</li>
<li>Bray-Curtis distance (a quantitative measure of community dissimilarity)</li>
<li>unweighted UniFrac distance (a qualitative measure of community dissimilarity that incorporates phylogenetic relationships between the features)</li>
<li>weighted UniFrac distance (a quantitative measure of community dissimilarity that incorporates phylogenetic relationships between the features)</li>
</ul>
</div></blockquote>
<p>The only parameter that needs to be provided to this script is <code class="docutils literal"><span class="pre">--p-counts-per-sample</span></code>, which is the even sampling (i.e. rarefaction) depth. Because most diversity metrics are sensitive to different sampling depths across different samples, this script will randomly subsample the counts from each sample to the value provided for this parameter. For example, if you provide <code class="docutils literal"><span class="pre">--p-counts-per-sample</span> <span class="pre">500</span></code>, this step will subsample the counts in each sample without replacement so that each sample in the resulting table has a total count of 500. If the total count for any sample(s) are smaller than this value, those samples will be dropped from the diversity analysis. Choosing this value is tricky. We recommend making your choice by reviewing the information presented in the <code class="docutils literal"><span class="pre">table.qzv</span></code> file that was created above and choosing a value that is as high as possible (so you retain more sequences per sample) while excluding as few samples as possible.</p>
<div class="question admonition" id="question-1">
<p class="first admonition-title">Question</p>
<p class="last">View the <code class="docutils literal"><span class="pre">table.qzv</span></code> artifact. What value would you choose to pass for the <code class="docutils literal"><span class="pre">--p-counts-per-sample</span></code>? How many samples will be excluded from your analysis based on this choice? Approximately how many total sequences will you be analyzing in the <code class="docutils literal"><span class="pre">core-metrics</span></code> command?</p>
</div>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime diversity core-metrics --i-phylogeny rooted-tree.qza --i-table table.qza --p-counts-per-sample <span class="m">1441</span> --output-dir cm1441
</pre></div>
</div>
<p>Here we set the <code class="docutils literal"><span class="pre">--p-counts-per-sample</span></code> parameter to 1441. This value was chosen here because it&#8217;s nearly the same number of sequences as the next few samples, and because it is the lowest value it will allow us to retain all of our samples. In many Illumina runs however you&#8217;ll observe a few samples that have much lower sequence counts (on the order of tens or a couple of hundred samples) - you will typically want to exclude those from the analysis by choosing a larger value.</p>
<p>After computing diversity metrics, we can begin to explore the microbial composition of the samples in the context of the sample metadata. This information is present in the <a class="reference external" href="https://docs.google.com/spreadsheets/d/1FXHtTmvw1gM4oUMbRdwQIEOZJlhFGeMNUvZmuEFqpps/edit?usp=sharing">sample metadata</a> file that was downloaded earlier (<code class="docutils literal"><span class="pre">sample-metadata.tsv</span></code>).</p>
<p>We&#8217;ll first test for associations between discrete metadata categories and alpha diversity data. We&#8217;ll do that here for the Faith Phylogenetic Diversity (a measure of community richness) and evenness metrics.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime diversity alpha-group-significance --i-alpha-diversity cm1441/faith_pd_vector.qza --m-metadata-file sample-metadata.tsv  --o-visualization cm1441/faith-pd-group-significance

qiime diversity alpha-group-significance --i-alpha-diversity cm1441/evenness_vector.qza --m-metadata-file sample-metadata.tsv  --o-visualization cm1441/evenness-group-significance
</pre></div>
</div>
<p><strong>faith-pd-group-significance</strong>: <a class="reference external" href="../../_viz/faith-pd-group-significance/09503df5-547c-489c-bfea-d7ffefdaa40f/data/index.html">visualization</a>, <a class="reference external" href="../../_viz/faith-pd-group-significance.qzv">qzv</a></p>
<p><strong>evenness-group-significance</strong>: <a class="reference external" href="../../_viz/evenness-group-significance/a7ec4e65-85d8-4433-97a6-8d9ae970f451/data/index.html">visualization</a>, <a class="reference external" href="../../_viz/evenness-group-significance.qzv">qzv</a></p>
<div class="question admonition" id="question-2">
<p class="first admonition-title">Question</p>
<p class="last">What discrete sample metadata categories are most strongly associated with the differences in microbial community <strong>richness</strong>? Are these differences statistically significant?</p>
</div>
<div class="question admonition" id="question-3">
<p class="first admonition-title">Question</p>
<p class="last">What discrete sample metadata categories are most strongly associated with the differences in microbial community <strong>evenness</strong>? Are these differences statistically significant?</p>
</div>
<p>Next, we&#8217;ll test for associations between alpha diversity metrics and continuous sample metadata (such as pH or elevation). We can do this running the following two commands, which will support analysis of Faith&#8217;s Phylogenetic Diversity metric and evenness in the context of our continuous metadata. Run these commands and view the resulting artifacts.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime diversity alpha-correlation --i-alpha-diversity cm1441/faith_pd_vector.qza --m-metadata-file sample-metadata.tsv  --o-visualization cm1441/faith-pd-correlation

qiime diversity alpha-correlation --i-alpha-diversity cm1441/evenness_vector.qza --m-metadata-file sample-metadata.tsv  --o-visualization cm1441/evenness-correlation
</pre></div>
</div>
<p><strong>faith-pd-correlation</strong>: <a class="reference external" href="../../_viz/faith-pd-correlation/db2c6b0c-c6f1-481d-8a29-dcd03744f54d/data/index.html">visualization</a>, <a class="reference external" href="../../_viz/faith-pd-correlation.qzv">qzv</a></p>
<p><strong>evenness-correlation</strong>: <a class="reference external" href="../../_viz/evenness-correlation/755b35d3-6a08-4f5c-8934-1a6b0ab45712/data/index.html">visualization</a>, <a class="reference external" href="../../_viz/evenness-correlation.qzv">qzv</a></p>
<div class="question admonition" id="question-4">
<p class="first admonition-title">Question</p>
<p class="last">What do you conclude about the associations between continuous sample metadata and the richness and evenness of these samples?</p>
</div>
<p>Next we&#8217;ll analyze sample composition in the context of discrete metadata using PERMANOVA (first described in <a class="reference external" href="http://onlinelibrary.wiley.com/doi/10.1111/j.1442-9993.2001.01070.pp.x/full">Anderson (2001)</a>) using the <code class="docutils literal"><span class="pre">beta-group-significance</span></code> command. The following commands will test whether distances between samples within a group, such as samples from the same body site (e.g., skin or gut), are more similar to each other then they are to samples from a different group. This command can be slow to run since it is based on permutation tests, so unlike the previous commands we&#8217;ll run this on specific categories of metadata that we&#8217;re interested in exploring, rather than all metadata categories that it&#8217;s applicable to. Here we&#8217;ll apply this to our unweighted UniFrac distances, using two sample metadata categories, as follows.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime diversity beta-group-significance --i-distance-matrix cm1441/unweighted_unifrac_distance_matrix.qza --m-metadata-file sample-metadata.tsv --m-metadata-category SampleType --o-visualization cm1441/unweighted-unifrac-sample-type-significance

qiime diversity beta-group-significance --i-distance-matrix cm1441/unweighted_unifrac_distance_matrix.qza --m-metadata-file sample-metadata.tsv --m-metadata-category Subject --o-visualization cm1441/unweighted-unifrac-subject-group-significance
</pre></div>
</div>
<p><strong>unweighted-unifrac-sample-type-significance</strong>: <a class="reference external" href="../../_viz/unweighted-unifrac-sample-type-significance/b50e2972-c4dc-4ce4-9625-b635da8fda5e/data/index.html">visualization</a>, <a class="reference external" href="../../_viz/unweighted-unifrac-sample-type-significance.qzv">qzv</a></p>
<p><strong>unweighted-unifrac-subject-group-significance</strong>: <a class="reference external" href="../../_viz/unweighted-unifrac-subject-group-significance/d7490128-4ec8-4bb8-80a3-c4aaa1238361/data/index.html">visualization</a>, <a class="reference external" href="../../_viz/unweighted-unifrac-subject-group-significance.qzv">qzv</a></p>
<div class="question admonition" id="question-5">
<p class="first admonition-title">Question</p>
<p class="last">Are the associations between subjects and differences in microbial composition statistically significant? How about sample types? What sample types appear to be most different from each other?</p>
</div>
<p>Finally, we&#8217;ll explore associations between the microbial composition of the samples and continuous sample metadata using bioenv (originally described in <a class="reference external" href="http://www.int-res.com/articles/meps/92/m092p205.pdf">Clarke and Ainsworth (1993)</a>). This approach tests for associations of pairwise distances between sample microbial composition (a measure of beta diversity) and sample metadata (for example, the matrix of Bray-Curtis distances between samples and the matrix of absolute differences in pH between samples). A powerful feature of this method is that it explores combinations of sample metadata to see which groups of metadata differences are most strongly associated with the observed microbial differences between samples. You can apply bioenv to the unweighted UniFrac distances and Bray-Curtis distances between the samples, respectively, as follows. After running these commands, open the resulting visualizations.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime diversity bioenv --i-distance-matrix cm1441/unweighted_unifrac_distance_matrix.qza --m-metadata-file sample-metadata.tsv --o-visualization cm1441/unweighted-unifrac-bioenv

qiime diversity bioenv --i-distance-matrix cm1441/bray_curtis_distance_matrix.qza --m-metadata-file sample-metadata.tsv --o-visualization cm1441/bray-curtis-bioenv
</pre></div>
</div>
<p><strong>unweighted-unifrac-bioenv</strong>: <a class="reference external" href="../../_viz/unweighted-unifrac-bioenv/b29ee241-2cf2-4849-9828-1d46252b74a4/data/index.html">visualization</a>, <a class="reference external" href="../../_viz/unweighted-unifrac-bioenv.qzv">qzv</a></p>
<p><strong>bray-curtis-bioenv</strong>: <a class="reference external" href="../../_viz/bray-curtis-bioenv/d927225c-e240-4753-90c3-c20bd2f66225/data/index.html">visualization</a>, <a class="reference external" href="../../_viz/bray-curtis-bioenv.qzv">qzv</a></p>
<div class="question admonition" id="question-6">
<p class="first admonition-title">Question</p>
<p class="last">What sample metadata or combinations of sample metadata are most strongly associated with the differences in microbial composition of the samples? How strong are these correlations?</p>
</div>
<p>Finally, ordination is a popular approach for exploring microbial community composition in the context of sample metadata. We can use the <a class="reference external" href="http://emperor.microbio.me">Emperor</a> tool to explore principal coordinates (PCoA) plots in the context of sample metadata. PCoA is run as part of the <code class="docutils literal"><span class="pre">core-metrics</span></code> command, so we can generate these plots for unweighted UniFrac and Bray-Curtis as follows. The <code class="docutils literal"><span class="pre">--p-custom-axis</span></code> parameter that we pass here is very useful for exploring temporal data. The resulting plot will contain axes for principal coordinate 1 (labelled <code class="docutils literal"><span class="pre">0</span></code>), principal coordinate 1 (labelled <code class="docutils literal"><span class="pre">1</span></code>), and days since the experiment start. This is useful for exploring how the samples change over time.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime emperor plot --i-pcoa cm1441/unweighted_unifrac_pcoa_results.qza --o-visualization cm1441/unweighted-unifrac-emperor --m-metadata-file sample-metadata.tsv --p-custom-axis DaysSinceExperimentStart

qiime emperor plot --i-pcoa cm1441/bray_curtis_pcoa_results.qza --o-visualization cm1441/bray-curtis-emperor --m-metadata-file sample-metadata.tsv --p-custom-axis DaysSinceExperimentStart
</pre></div>
</div>
<p><strong>unweighted-unifrac-emperor</strong>: <a class="reference external" href="../../_viz/unweighted-unifrac-emperor/a3c6f11c-8e40-4a85-9d70-598dc335a038/data/index.html">visualization</a>, <a class="reference external" href="../../_viz/unweighted-unifrac-emperor.qzv">qzv</a></p>
<p><strong>bray-curtis-emperor</strong>: <a class="reference external" href="../../_viz/bray-curtis-emperor/873f10c0-c301-47bc-a42c-8bbb00d9938a/data/index.html">visualization</a>, <a class="reference external" href="../../_viz/bray-curtis-emperor.qzv">qzv</a></p>
<div class="question admonition" id="question-7">
<p class="first admonition-title">Question</p>
<p class="last">Do the Emperor plots support the other beta diversity analyses we&#8217;ve performed here? (Hint: Experiment with coloring points by different metadata.)</p>
</div>
<div class="question admonition" id="question-8">
<p class="first admonition-title">Question</p>
<p class="last">What differences do you observe between the unweighted UniFrac and Bray-Curtis PCoA plots?</p>
</div>
</div>
<div class="section" id="taxonomic-analysis">
<h2>Taxonomic analysis<a class="headerlink" href="#taxonomic-analysis" title="Permalink to this headline">¶</a></h2>
<p>In the next sections we&#8217;ll begin to explore the taxonomic composition of the samples, and again relate that to sample metadata. The first step in this process is to assign taxonomy to the sequences in our <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> artifact. We&#8217;ll do that using a Naive Bayes classifier with the <code class="docutils literal"><span class="pre">q2-feature-classifier</span></code> plugin. This classifier was trained on the Greengenes 13_8 99% OTUs, where the sequences have been trimmed to only include the region of the 16S that was sequenced in this analysis (the V4 region, bound by the 515F/806R primer pair). We&#8217;ll download and apply the pre-trained classifier here because training this classifier can be slow, but it is easy to train Naive Bayes and other classifiers on custom sequence collections using the <code class="docutils literal"><span class="pre">q2-feature-classifier</span></code> plugin. We&#8217;ll then apply this classifier to our sequences, and we can generate a visualization of the resulting mapping from sequence to taxonomy.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>curl -sLO http://data.qiime2.org/common/gg-13-8-99-515-806-nb-classifier.qza

qiime feature-classifier classify --i-classifier gg-13-8-99-515-806-nb-classifier.qza --i-reads rep-seqs.qza --o-classification taxonomy

qiime feature-table view-taxa-data --i-data taxonomy.qza --o-visualization taxonomy
</pre></div>
</div>
<p><strong>taxonomy</strong>: <a class="reference external" href="../../_viz/taxonomy/0b32cf69-34c7-4e16-aff3-e486d79409c9/data/index.html">visualization</a>, <a class="reference external" href="../../_viz/taxonomy.qzv">qzv</a></p>
<div class="question admonition" id="question-9">
<p class="first admonition-title">Question</p>
<p class="last">Recall that our <code class="docutils literal"><span class="pre">rep-seqs.qzv</span></code> artifact allows you to easily BLAST the sequence associated with each feature against the NCBI nt database. Using that artifact and the <code class="docutils literal"><span class="pre">taxonomy.qzv</span></code> artifact created here, compare the taxonomic assignments with the taxonomy of the best BLAST hit for a few features. How similar are the assignments? If they&#8217;re dissimilar, at what <em>taxonomic level</em> do they begin to differ (e.g., species, genus, family, ...)?</p>
</div>
<p>Next, we can view the taxonomic composition of our samples with interactive bar plots. Generate those plots with the following command and then open the visualization.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime taxa barplot --i-table table.qza --i-taxonomy taxonomy.qza --m-metadata-file sample-metadata.tsv --o-visualization taxa-bar-plots
</pre></div>
</div>
<p><strong>taxa-bar-plots</strong>: <a class="reference external" href="../../_viz/taxa-bar-plots/059297b8-2557-4cca-84da-281185238aa1/data/index.html">visualization</a>, <a class="reference external" href="../../_viz/taxa-bar-plots.qzv">qzv</a></p>
<div class="question admonition" id="question-10">
<p class="first admonition-title">Question</p>
<p class="last">Visualize the samples at <em>Level 2</em> (which corresponds to the phylum level in this analysis), and then sort the samples by SampleType, then by Subject, and then by DaysSinceExperimentStart. What are the dominant phyla in each in SampleType? Do you observe any consistent change across the two subjects between DaysSinceExperimentStart <code class="docutils literal"><span class="pre">0</span></code> and the later timepoints?</p>
</div>
</div>
<div class="section" id="differential-abundance-analysis">
<h2>Differential abundance analysis<a class="headerlink" href="#differential-abundance-analysis" title="Permalink to this headline">¶</a></h2>
<p>Finally, we can quantify the process of identifying taxa that are differentially abundance (or present in different abundances) across sample groups. We do that using ANCOM (<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/26028277">Mandal et al. (2015)</a>), which is implemented in the <code class="docutils literal"><span class="pre">q2-composition</span></code> plugin. ANCOM operates on a <code class="docutils literal"><span class="pre">FeatureTable[Composition]</span></code> artifact, which is based on relative frequencies of features on a per-sample basis, but cannot tolerate frequencies of zero. We work around this by adding a pseudocount of 1 to every count in our <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> table. We can run this on the <code class="docutils literal"><span class="pre">SampleType</span></code> category to determine what features differ in abundance across our sample types. This step may take about 5 minutes to complete.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime composition add-pseudocount --i-table table.qza --o-composition-table comp-table

qiime composition ancom --i-table comp-table.qza --m-metadata-file sample-metadata.tsv --m-metadata-category SampleType --o-visualization ancom-SampleType
</pre></div>
</div>
<p><strong>ancom-SampleType</strong>: <a class="reference external" href="../../_viz/ancom-SampleType/f2ee210b-1a25-478f-bcfc-37424610bad1/data/index.html">visualization</a>, <a class="reference external" href="../../_viz/ancom-SampleType.qzv">qzv</a></p>
<div class="question admonition" id="question-11">
<p class="first admonition-title">Question</p>
<p class="last">What features differ in abundance across SampleType? What groups are they most and least abundant in? What are some the taxonomies of some of these features? (To answer that last question you&#8217;ll need to refer to a visualization that we generated earlier in this tutorial.)</p>
</div>
<p>We&#8217;re also often interested in performing a differential abundance test at a specific taxonomic level. To do this, we can collapse the features in our <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> at the taxonomic level of interest, and then re-run the above steps.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime taxa collapse --i-table table.qza --i-taxonomy taxonomy.qza --p-level <span class="m">2</span> --o-collapsed-table table-l2

qiime composition add-pseudocount --i-table table-l2.qza --o-composition-table comp-table-l2

qiime composition ancom --i-table comp-table-l2.qza --m-metadata-file sample-metadata.tsv --m-metadata-category SampleType --o-visualization l2-ancom-SampleType
</pre></div>
</div>
<p><strong>l2-ancom-SampleType</strong>: <a class="reference external" href="../../_viz/l2-ancom-SampleType/7a44482c-fe21-4c35-91d3-d74c3aaba728/data/index.html">visualization</a>, <a class="reference external" href="../../_viz/l2-ancom-SampleType.qzv">qzv</a></p>
<div class="question admonition" id="question-12">
<p class="first admonition-title">Question</p>
<p class="last">What phyla differ in abundance across SampleType? How does this align with what you observed in the <code class="docutils literal"><span class="pre">taxa-bar-plots.qza</span></code> visualization that was generated above?</p>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="../index.html">
              <img class="logo" src="../_static/logo.png" alt="Logo"/>
            </a></p>
  <h3><a href="../index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">&#8220;Moving Pictures&#8221; tutorial</a><ul>
<li><a class="reference internal" href="#prepare-for-the-analysis">Prepare for the analysis</a></li>
<li><a class="reference internal" href="#demultiplexing-sequences">Demultiplexing sequences</a></li>
<li><a class="reference internal" href="#sequence-quality-control">Sequence quality control</a></li>
<li><a class="reference internal" href="#generate-a-tree-for-phylogenetic-diversity-analyses">Generate a tree for phylogenetic diversity analyses</a></li>
<li><a class="reference internal" href="#alpha-and-beta-diversity-analysis">Alpha and beta diversity analysis</a></li>
<li><a class="reference internal" href="#taxonomic-analysis">Taxonomic analysis</a></li>
<li><a class="reference internal" href="#differential-abundance-analysis">Differential abundance analysis</a></li>
</ul>
</li>
</ul>
<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
  <li><a href="index.html">Tutorials</a><ul>
      <li>Previous: <a href="index.html" title="previous chapter">Tutorials</a></li>
      <li>Next: <a href="fmt.html" title="next chapter">Fecal microbiota transplant (FMT) study: an exercise</a></li>
  </ul></li>
  </ul></li>
</ul>
</div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2016, QIIME 2 Development Team.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.4.6</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.9</a>
      
      |
      <a href="../_sources/tutorials/moving-pictures.txt"
          rel="nofollow">Page source</a>
    </div>

    
    <a href="https://github.com/qiime2/" class="github">
        <img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_darkblue_121621.png" alt="Fork me on GitHub"  class="github"/>
    </a>
    

    
  </body>
</html>